### DISCUSSION TITLE
Video support

### DISCUSSION THREAD

**Discussion by steve1977** [RandomNinjaAtk/docker-lidarr-extended#29] [Ideas]
Moving our discussion from Discord to this forum. I am convinced of your approach to musicvideos (IMVDB, YT&T links from MBZ). Happy to further discuss here, hear other views, and also test.

And a small other topic: Possible to set the audio download to D, L, both and none. Currently, it doesn't seem possible to de-activate? Same for a potential future video support function.

**Comment by RandomNinjaAtk** [comment]
If you set the dlClientSource to anything other than the provided options, it should act as if you set it to "none", but I have not tested that, give it a try

**Comment by steve1977** [comment]
Thanks, will try.

Was thinking again about the MV feature. At second thought, your very original script would be perfect and you may want to limit it to imvdb links and YT links from mbz. Once you allow mbz links to other sources (tidal, etc), you’ll surely create duplicates as there won’t be any matching between imvdb links and other links (which you are doing for mbz links based on the yt url.

This may also make it easier as you had this running last year quite well.

**Comment by RandomNinjaAtk** [comment]
If you really want to, you can start testing the new video script. You'll need to manually execute it.

`bash /config/extended/scripts/video.sh`

Its very much a work-in-progress and likely to change. It only uses musicbrainz as a source. So I don't expect it to pickup a lot until people work together to populate musicbrainz...

I've only tested it with a single artist (https://musicbrainz.org/artist/adc0f033-95c2-4e0b-87bc-c23ed3f26ce6/releases).

**Comment by steve1977** [comment]
Thanks. Not yet this weekend, but will try out next.

Did you come to the conclusion not to support IMVDB anymore? In the end, I found your logic very good to support IMVDB as it is still being maintained / added? Plus it wouldn't create duplicates with MBZ if you look out for YT links.

**Reply by RandomNinjaAtk** [reply]
Its more difficult to maintain and support multiple sources and because IMVDB cannot be user updated/maintained, I think it would be easier to streamline and focus on only one source for data.

So I think only using musicbrainz is best. It's not ideal based on the data that is currently there, because a lot of it needs to be updated and have some much needed TLC. But I think it will overall be the best long term solution.

I just need to figure out a naming convention and write a guide for adding/updating Musicbrainz to try to help standardize it.

**Comment by steve1977** [comment]
Thanks. I don't think MBZ will all our efforts will ever get us any meaningful scale in terms of YT videos. IMVDB is formally linked to MBZ, so the community can make efforts to further link IMVDB links to MBZ. IMVDB no longer accepts user contributions, but remains very active in adding videos. Dupes can be avoided by keeping log for YT links that have been previously downloaded.

Naming convention could follow the MBZ artist name and the YT name, which could be an alternate method to avoid dupes (i.e., instead by URL, you could avoid dupes based on "MBZ artist - "YT file name". To avoid confusing Plex or others, you could use MP4/M4V tags with the MBZ artist name and the IMVDB video name. This way, you'd have the best of both worlds and a better way for dupe avoidance.

I am basically using the arguments you used when we discussed on Discord why IMVDB isn't as bad as I thought it would. After you challenged me, I revisited IMVDB and indeed realized that it is a pretty good site with great quality links and no real issues.

Maybe you want to reconsider IMVDB and basically get back to what you had in the original script, which was quite powerful!

Thoughts?

**Reply by RandomNinjaAtk** [reply]
Don't think I'm going to change it because it requires to much effort to maintain and it over complicates the overall process...

**Comment by steve1977** [comment]
In this case, why don't you go with IMDVB-only? No maintenance, simple, clean, and reasonably well maintained. Still active & adding, fully integrated with MBZ.

**Reply by RandomNinjaAtk** [reply]
That wouldn't support the use of alternate sources, such as tidal

**Comment by steve1977** [comment]
That's true, but mixing YT and Tidal anyways create a massive mess as there is basically no way you could eliminate dupes. Let's assume I'd add my favorite MVs to MBZ with Tidal links. At the same time, you'd do the same with YT links, we'd create dupes, which will basically be impossible to sort out.

You know I was originally very excited of you adding Tidal links and the quality of Tidal MVs. However, then the reality check came in and that confirmed that IMDVB is still by far the best source. And if you were to allow some form of community support, maybe adding YT links from MBZ isn't too bad.

Unsolved is the naming convention, but key likely to use the MBZ artist, which you could do in either way.

**Reply by RandomNinjaAtk** [reply]
Dupes is not an issue with both, unless users are creating duplicate MBZ records, instead of adding both links to the same record. 

Example Record with no issues: https://musicbrainz.org/recording/56d0c749-148b-4197-8620-f740e5feca0f

But if a user created 2 records, one with each link, you could possibly have a dupe, but only if the record titles are spelled differently, since there is a built in filename check also...

**Comment by steve1977** [comment]
I got the theory, but the reality is that there would be many recordings for the same song. I would not call this a strength of imvdb. And even less if lidarr users would add their personal MV preferences.

**Reply by RandomNinjaAtk** [reply]
This only pulls videos that are considered official, so unless users are deliberately miss labeling things, like labeling a recording as official when it is not, it shouldn't be an issue. And if users are intentionally doing that, there isn't really any way to prevent it other then it being noticed and then making edits to the MBZ database to correct it.

It's a community managed service/database, be part of the community ;)

**Comment by steve1977** [comment]
I am surely part of the MBZ community. Not sure whether I can pull any facts, but I must have done 10k+ edits to MBZ by now. For example, I must added more than 1,000 IMDVB links to MBZ. Plus of course album additions, changes, etc.

What do you mean pulling only "official videos"? How does it differentiate between official and non-official videos?

My concern with the current thinking is:

* Right now, it doesn't really do anything as close to no video links are in MBZ

* If people like myself would add MV links to MBZ. Some may add Tidal and others may add YT links. Some may add low-quality or even random tv recordings. Plus Tidal video links may expire & change, which makes things messy

I see three potentially better options:

* IMVDB only: the reason why they closed the community may be that they cannot control it (which MBZ may also not be able to do), but the database is actually very strong. I revisited it and it is basically the highest quality and best videos you can think of

* IMVDB plus seleced YT: basically what you had in the past, but with stronger restrictions what would be added from MBZ links

* Creating a new database: Not sure you know how to do or have appetite, but you could also initiate an alternative to IMVDB that is community-driven and serves as single source of truth to MVs. Basically recreating to what IMVDB does, but keeping it open

**Reply by RandomNinjaAtk** [reply]
"What do you mean pulling only "official videos"? How does it differentiate between official and non-official videos?"
[Q:A Extended-Video-Script-Information](https://github.com/RandomNinjaAtk/docker-lidarr-extended/wiki/Extended-Video-Script-Information#q-their-is-a-lot-of-music-videos-available-online-youtube-or-tidal-yet-the-script-cannot-find-them-what-can-i-do)


Your third better option sounds familiar... I think it's called MusicBrainz... it's a community-driven database...

Edit:
"If people like myself would add MV links to MBZ. Some may add Tidal and others may add YT links. Some may add low-quality or even random tv recordings. Plus Tidal video links may expire & change, which makes things messy"

This can be a problem for every solution....

"Right now, it doesn't really do anything as close to no video links are in MBZ"

This would be a problem for your proposed third better solution...


The best solution is Musicbrainz, your third option is basically asking to recreate what MusicBrainz already does, and we're not going to reinvent the wheel by creating a second MusicBrainz...

**Comment by steve1977** [comment]
I don't think IMVDB recreated MBZ. I think MBZ has its purpose and IMVDB has its purpose. The two are linked, which is great! So if you speak about "recreation", I am basically suggesting to either use IMVDB or recreate it.

The current implementation is a barely work-around to make MBZ mimic IMVDB (by adding "official" and finding out ways to avoid dupes Tidal and YT). IMVDB doesn't need any of these work-arounds and is designed for MVs.

In fact, IMVDB is quite perfect for our purpose and an excellent rich source of data. The only short-coming is that they closed community contributions (which is unclear whether that's temporary or for good). So either live with a more restrictive IMVDB or create a better IMVDB for MVs.

Let me also try contact the IMVDB developers again.

**Comment by steve1977** [comment]
Just to give you some stats. IMVDB has perfectly curated MVs for below, which are perfectly named, without dupes, only official videos, etc. The only issue is that community may no longer be invited to contribute, which I am contacting the developers about.

DB Stats
92,769 Music Videos
458,398 Artists, Companies, and People
263,725 Credits

**Reply by Protektor-Desura** [reply]
I've tried to add stuff to the database but the site basically gave me the finger for trying to keep it updated. :(

**Comment by steve1977** [comment]
I know I know... Same issue I was and am facing...

I'm just saying that it is an amazingly well curated database, which seen frequent updates (with detailed info) and no dupes. It is quite impressive what they are doing and the pace of it.

The only short-coming is that they basically seem not to allow any community entries. At least not from people like you or me...

**Comment by steve1977** [comment]
Fwiw, even if imvdb continues to ignore their community, it is actively and well maintained. The detail, accuracy and speed of adding new MVs is quite impressive. And no dupes or otherwise annoying videos.

Even if we were to all start adding to MBZ, we'd never get there and will argue around us adding our favorites mis-using the work-around with "official".

So, I see only benefits in you reconsidering your stance and add imvdb support (using MBZ for links and artist naming). Even the imvdb MV names are quite perfect!

**Comment by RandomNinjaAtk** [comment]
IMVDB was implemented in the latest version.
